packages/components/src/templates/next/components/internal/Wizgov/WizgovWidgetClient.tsx

```diff
@@ -0,0 +1,45 @@
+"use client"
+
+import { useEffect } from "react"
+
+import type { WizgovWidgetClientProps } from "~/interfaces"
+
+const reloadWizgovScript = (
+  scriptUrl: WizgovWidgetClientProps["scriptUrl"],
+) => {
+  const scriptId = "isomer-wizgov-script"
+
+  const existingScriptTag = document.getElementById(scriptId)
+  if (existingScriptTag) {
+    existingScriptTag.remove()
+  }
+
+  const scriptTag = document.createElement("script")
+  scriptTag.id = scriptId
+  scriptTag.async = true
+  scriptTag.type = "text/javascript"
+  scriptTag.src = scriptUrl
+  scriptTag.referrerPolicy = "origin"
+  document.body.appendChild(scriptTag)
+}
+
+export const WizgovWidgetClient = (props: WizgovWidgetClientProps) => {
+  const { "data-agency": dataAgency, scriptUrl } = props
+
+  useEffect(() => {
+    // to not render during static site generation on the server
+    if (typeof window === "undefined") return
+
+    // Only inject the script after everything else has finished loading
+    // This is to replicate Next.js lazyOnload behaviour (as recommended for widgets)
+    if (document.readyState === "complete") {
+      reloadWizgovScript(scriptUrl)
+    } else {
+      window.addEventListener("load", () => reloadWizgovScript(scriptUrl))
+      return () =>
+        window.removeEventListener("load", () => reloadWizgovScript(scriptUrl))
+    }
```

Comment:
question(non-blocking): out of curiosity, is the Wizgov script able to load multiple times but only still show one single icon, [similar to WOGAA](https://github.com/opengovsg/isomer/blob/main/packages/components/src/templates/next/components/internal/Wogaa/Wogaa.tsx)?

Asking cos using the ScriptComponent (next/script) would be more ideal (not sure if more optimised, never checked) so if we can use that instead of our own useEffect would be good.